### PR TITLE
AB2D-5797 Copy Opt-out file from BFD S3 to AB2D hosted S3 for backup

### PR DESCRIPTION
**Pull request by smirnovaae**
## 🎫 Ticket

https://jira.cms.gov/browse/AB2D-5797

## 🛠 Changes

Added OptOutS3.copyFilesFromS3() to copy files from BFD S3 to AB2D S3

## ✅ Acceptance Validation

Tested manually using temporary S3 bucket on AB2D dev environment.

Screenshot 2023-10-23 at 10 57 05 AM

newoptOutDummy.txt is a copy of optOutDummy.txt

PS: I will remove `new` from file name when we switch on AB2D S3 on prod environment

## 🔒 Security Implications

- [ ] This PR adds a new software dependency or dependencies.
- [ ] This PR modifies or invalidates one or more of our security controls.
- [ ] This PR stores or transmits data that was not stored or transmitted before.
- [ ] This PR requires additional review of its security implications for other reasons.

If any security implications apply, add Jason Ashbaugh (GitHub username: StewGoin) as a reviewer and do not merge this PR without his approval.

### CODE REVIEW

File: optout/src/main/java/gov/cms/ab2d/optout/OptOutS3.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+import com.amazonaws.services.s3.AmazonS3ClientBuilder;
+import com.amazonaws.services.s3.iterable.S3Objects;
+import com.amazonaws.services.s3.model.*;
+
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.util.concurrent.CountDownLatch;
+
+
+public class OptOutS3 implements Runnable {
```
Comment (meliGuzman) [inline optout/src/main/java/gov/cms/ab2d/optout/OptOutS3.java:16]: Small comment, non-blocking. I recommend taking advantage of Lombok if you'd like. I believe this project uses it already in different classes. You could add these annotations: @NoArgsConstructor, @RequiredArgsConstructor, @AllArgsConstructor, and @Builder for this class and take advantage of them. Feel free to ignore though! just a style choice

https://projectlombok.org/features/constructor
https://projectlombok.org/features/Builder

Comment (smirnovaae) [reply optout/src/main/java/gov/cms/ab2d/optout/OptOutS3.java:16]: Thank you for your comment!
We use lombok in the ab2d repository, but we do not use it in ab2d-lambdas and even exclude it from dependencies when we import something from AB2D-Libs.
The Lombok library increases the jar size and AWS has a size quota for lambda. In other words, we want to import only the necessary libraries.

Comment (meliGuzman) [reply optout/src/main/java/gov/cms/ab2d/optout/OptOutS3.java:16]: Makes sense and thank you for explaining, @smirnovaae!

**Comment by Sadibhatla** [comment]
Follow up tickets created for Multiple files processing and  Moving s3 bucket names to Secret manager..  Looks good to me. Lets address these in the following versions!
Thank you

**Review by Sadibhatla — APPROVED** [review]

**Review by Rwolfe-Nava — APPROVED** [review]
Looks good to me

**Pull request merged**